Source/ASDisplayNode.mm

```diff
@@ -3934,11 +3934,13 @@ - (ASEventLog *)eventLog
     [result addObject:@{ @"alpha" : @(_pendingViewState.alpha) }];
     [result addObject:@{ @"frame" : [NSValue valueWithCGRect:_pendingViewState.frame] }];
   }
-  
-  // Check supernode so that if we are cell node we don't find self.
-  ASCellNode *cellNode = ASDisplayNodeFindFirstSupernodeOfClass(self.supernode, [ASCellNode class]);
-  if (cellNode != nil) {
-    [result addObject:@{ @"cellNode" : ASObjectDescriptionMakeTiny(cellNode) }];
+
+  if (ASDisplayNodeThreadIsMain()) {
+    // Check supernode so that if we are cell node we don't find self.
+    ASCellNode *cellNode = ASDisplayNodeFindFirstSupernodeOfClass(self.supernode, [ASCellNode class]);
+    if (cellNode != nil) {
+      [result addObject:@{ @"cellNode" : ASObjectDescriptionMakeTiny(cellNode) }];
+    }
   }
```

Comment:
Should we have an else here that mentions it couldn't look for the cell node because it was called off main?